plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/MachineParameters.hs

```diff
@@ -48,7 +48,9 @@ data MachineParameters machinecosts term (uni :: Type -> Type) (fun :: Type) =
 -- See Note [Inlining meanings of builtins].
 {-| This just uses 'toBuiltinsRuntime' function to convert a BuiltinCostModel to a BuiltinsRuntime. -}
 mkMachineParameters ::
-    ( -- In Cek.Internal we have `type instance UniOf (CekValue uni fun) = uni`, but we don't know that here.
+    ( -- WARNING: do not discharge the equality constraint as that causes GHC to fail to inline the
+      -- /caller/ of this function at /its/ call site. I.e. 'mkMachineParameters' continues to
```

Comment:
Actually it _is_ `mkMachineParameters` that fails to be inlined. In the core dump I'm seeing
```
immediateMachineParameters1
  = $smkMachineParameters UnliftingImmediate
```

Reply:
Oh indeed, I got confused. Thank you, will fix.